src/mrinufft/extras/field_map.py

```diff
@@ -0,0 +1,106 @@
+"""Field map generator module."""
+
+import numpy as np
+
+
+def make_b0map(shape, b0range=(-300, 300), mask=None):
+    """
+    Make radial B0 map.
+
+    Parameters
+    ----------
+    shape : tuple[int]
+        Matrix size. Only supports isotropic matrices.
+    b0range : tuple[float], optional
+        Frequency shift range in [Hz]. The default is (-300, 300).
+    mask : np.ndarray
+        Spatial support of the objec. If not provided,
+        build a radial mask with radius = 0.3 * shape
+
+    Returns
+    -------
+    np.ndarray
+        B0 map of shape (*shape) in [Hz],
+        with values included in (*b0range).
+    mask : np.ndarray, optional
+        Spatial support binary mask.
+
+    """
+    assert np.unique(shape).size, ValueError("Only isotropic matriex are supported.")
+    ndim = len(shape)
+    if ndim == 2:
+        radial_mask, fieldmap = _make_disk(shape)
+    elif ndim == 3:
+        radial_mask, fieldmap = _make_sphere(shape)
+    if mask is None:
+        mask = radial_mask
+
+    # build map
+    fieldmap *= mask
+    fieldmap = (b0range[1] - b0range[0]) * fieldmap / fieldmap.max() + b0range[0]  # Hz
+    fieldmap *= mask
```

Comment:
I don't think it's useful to reapply the binary mask. Do you agree? fieldmap should be already aligned with the binary mask given Line 38

Reply:
First masking is to restrict the rescaling to the region of interest (so that fieldmap.max() in line 40 would not refer to the FOV edges anymore). I put the second one to avoid having a constant b0 offset (b0range[0]) in the background. I can change though if you think this makes more sense!